From d0efa1ecc297701d5c72262a103c93f95e4137ca Mon Sep 17 00:00:00 2001
From: Numan Siddique <nusiddiq@redhat.com>
Date: Thu, 8 Dec 2016 16:48:00 +0530
Subject: [PATCH 4/4] ovn-ctl: Add remotes in 'Connection' table of NB and SB
 DBs

When starting ovsdb-server's for NB and SB dbs add the
remote connections in the 'Connection' tables if DB_NB_DEFAULT_REMOTE
is not set.

Signed-off-by: Numan Siddique <nusiddiq@redhat.com>
---
 ovn/utilities/ovn-ctl | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/ovn/utilities/ovn-ctl b/ovn/utilities/ovn-ctl
index 4dade90..2df60fb 100755
--- a/ovn/utilities/ovn-ctl
+++ b/ovn/utilities/ovn-ctl
@@ -116,6 +116,20 @@ start_ovsdb () {
 
         $@ $DB_NB_FILE
         ovn-nbctl init
+
+        if test X"$DB_NB_DEFAULT_REMOTE" = Xno; then
+            conn_info=`ovn-nbctl  find Connection target="ptcp\:$DB_NB_PORT\:$DB_NB_ADDR"`
+            conn_uuid=`echo $conn_info | awk '{print $3'}`
+            if test X"$conn_uuid" = X; then
+                ovn-nbctl set-connection "ptcp:$DB_NB_PORT:$DB_NB_ADDR"
+                conn_info=`ovn-nbctl  find Connection target="ptcp\:$DB_NB_PORT\:$DB_NB_ADDR"`
+                conn_uuid=`echo $conn_info | awk '{print $3'}`
+            fi
+
+            if test X"$DB_NB_INACTIVITY_PROBE" != X; then
+                ovn-nbctl set Connection $conn_uuid inactivity_probe=$DB_NB_INACTIVITY_PROBE
+            fi
+        fi
     fi
 
     # Check and eventually start ovsdb-server for Southbound DB
@@ -147,6 +161,20 @@ start_ovsdb () {
 
         $@ $DB_SB_FILE
         ovn-sbctl init
+
+        if test X"$DB_SB_DEFAULT_REMOTE" = Xno; then
+            conn_info=`ovn-sbctl  find Connection target="ptcp\:$DB_SB_PORT\:$DB_SB_ADDR"`
+            conn_uuid=`echo $conn_info | awk '{print $3'}`
+            if test X"$conn_uuid" = X; then
+                ovn-sbctl set-connection "ptcp:$DB_SB_PORT:$DB_SB_ADDR"
+                conn_info=`ovn-sbctl  find Connection target="ptcp\:$DB_SB_PORT\:$DB_SB_ADDR"`
+                conn_uuid=`echo $conn_info | awk '{print $3'}`
+            fi
+
+            if test X"$DB_SB_INACTIVITY_PROBE" != X; then
+                ovn-sbctl set Connection $conn_uuid inactivity_probe=$DB_SB_INACTIVITY_PROBE
+            fi
+        fi
     fi
 }
 
@@ -338,6 +366,8 @@ set_defaults () {
 
     DB_SB_DEFAULT_REMOTE="no"
     DB_NB_DEFAULT_REMOTE="no"
+    DB_NB_INACTIVITY_PROBE=""
+    DB_SB_INACTIVITY_PROBE=""
 }
 
 set_option () {
@@ -413,10 +443,12 @@ File location options:
   --db-nb-sync-from-port=PORT OVN Northbound active db tcp port (default: $DB_NB_SYNC_FROM_PORT)
   --db-nb-sync-from-proto=PROTO OVN Northbound active db transport (default: $DB_NB_SYNC_FROM_PROTO)
   --db-nb-default-remote=yes|no Use OVN Northbound default remote connection (default: $DB_NB_DEFAULT_REMOTE)
+  --db-nb-inactivity-probe=TIME Set inactivity probe (in msec) for NB remote (default:$DB_NB_INACTIVITY_PROBE)
   --db-sb-sync-from-addr=ADDR OVN Southbound active db tcp address (default: $DB_SB_SYNC_FROM_ADDR)
   --db-sb-sync-from-port=ADDR OVN Southbound active db tcp port (default: $DB_SB_SYNC_FROM_PORT)
   --db-sb-sync-from-proto=PROTO OVN Southbound active db transport (default: $DB_SB_SYNC_FROM_PROTO)
   --db-sb-default-remote=yes|no Use OVN Southbound default remote connection (default: $DB_SB_DEFAULT_REMOTE)
+  --db-sb-inactivity-probe=TIME Set inactivity probe (in msec) for SB remote (default: $DB_SB_INACTIVITY_PROBE)
 
 Default directories with "configure" option and environment variable override:
   logs: /usr/local/var/log/openvswitch (--with-logdir, OVS_LOGDIR)
-- 
2.9.3

